crc/services/user_service.py

```diff
@@ -20,8 +20,12 @@ def has_user():
 
     # Returns true if the current user is an admin.
     @staticmethod
-    def user_is_admin():
-        return UserService.has_user() and g.user.is_admin()
+    def user_is_admin(allow_admin_impersonate=False):
+        if allow_admin_impersonate:
+            user = UserService.current_user(allow_admin_impersonate)
+        else:
+            user = g.user
+        return UserService.has_user() and user.is_admin()
```

Comment:
feels like the check for UserService.has_user() should happen right away - to avoid the error on line 27 when there is no g.user